UnchainedLauncherGUI/src/App.xaml.cs

```diff
@@ -22,5 +30,53 @@ public App() : base() {
                 }
             }
         }
+        protected override void OnStartup(StartupEventArgs e) {
+            base.OnStartup(e);
+
+
+            // Init common dependencies
+            Octokit.GitHubClient githubClient = new Octokit.GitHubClient(new Octokit.ProductHeaderValue("UnchainedLauncher"));
+
+            var installationFinder = new Chivalry2InstallationFinder();
+            IUnchainedLauncherInstaller installer = new UnchainedLauncherInstaller(githubClient, Current.Shutdown);
+
+            // figure out if we need to install
+            var currentDirectory = Directory.GetParent(Assembly.GetExecutingAssembly().Location);
+            var needsInstallation = currentDirectory != null && !installationFinder.IsValidInstallation(currentDirectory);
+
+            // initialize the window
+            Window window = 
+                needsInstallation 
+                    ? InitializeInstallerWindow(installationFinder, installer) 
+                    : InitializeMainWindow(installationFinder, installer);
+
+            window.Show();
+        }
+
+        public static Window InitializeInstallerWindow(Chivalry2InstallationFinder installationFinder, IUnchainedLauncherInstaller installer) {
+            var installationSelectionVM = new InstallationSelectionPageViewModel(installationFinder);
+            var versionSelectionVM = new VersionSelectionPageViewModel(installer);
+            var installationLogVM = new InstallerLogPageViewModel(
+                installer,
+                () => from chiv2Installations in installationSelectionVM.Installations
+                      where chiv2Installations.IsSelected
+                      select chiv2Installations.Path
+                ,
+                () => versionSelectionVM.SelectedVersion!
+            );
+
+            ObservableCollection<IInstallerPageViewModel> installerPageViewModels = new ObservableCollection<IInstallerPageViewModel> {
+                installationSelectionVM,
+                versionSelectionVM,
+                installationLogVM
+            };
+
+            var installerWindowVM = new InstallerWindowViewModel(installerPageViewModels, installationSelectionVM.Installations);
+            return new InstallerWindow(installerWindowVM);
+        }
+
+        public static Window InitializeMainWindow(IChivalry2InstallationFinder installationFinder, IUnchainedLauncherInstaller installer) {
+            return new MainWindow(installationFinder, installer);
+        }
```

Comment:
These are probably the most consequential changes for the rest of the project @Knutschbert  and @ARoese 

`App.xaml.cs` is where you're supposed to initialize dependencies and your root DataContext.  Currently we do all that in the MainWindow constructor, but most of it should move here (and will soon after this PR is merged)